bin/plugin/commands/changelog.js

```diff
@@ -501,8 +635,36 @@ async function getChangelog( settings ) {
 			continue;
 		}
 
+		const featureGroups = groupBy( groupPullRequests, getIssueFeature );
 		changelog += '### ' + group + '\n\n';
-		groupEntries.forEach( ( entry ) => ( changelog += entry + '\n' ) );
+
+		Object.keys( featureGroups )
+			.sort( ( a, b ) => {
+				// sort "Unknown" to always be at the end
+				if ( a === 'Unknown' ) {
+					return 1;
+				} else if ( b === 'Unknown' ) {
+					return -1;
+				}
+				return featureGroups[ b ].length - featureGroups[ a ].length;
+			} )
+			.forEach( ( feature ) => {
+				const featureGroup = featureGroups[ feature ];
+				// Start new markdown list
+				changelog += '- ' + feature + '\n';
+				const featureGroupEntries = featureGroup
+					.map( getEntry )
+					.filter( Boolean );
+				featureGroupEntries.forEach( ( entry ) => {
+					// Strip feature name from entry if present.
+					entry = entry && entry.replace( `[${ feature } - `, '[' );
+
+					// Add a new bullet point to the list.
+					changelog += `  ${ entry }\n`;
+				} );
+				// End the markdown list.
+				changelog += '\n';
```

Comment:
The logic here (or at least some of it) could be extracted to a method. Moreover perhaps template strings could be used to improve legibility of building up the markdown rather than using concatenation.

Reply:
One for a followup